### PR TITLE
refactor(icons)!: Use a better logic to parse icons.

### PR DESCRIPTION
**Pull request by Jint-lzxy**
[\[View Diff Here\]](https://github.com/ayamir/nvimdots/compare/dc7ce68719cbc9d8e64dd6598a790562cae7b598..07828e61058dec041e61222156a4314a97401c87) _(Compared to #285)_.

This commit refactored the icon parsing logic. See the code for implementation detail. @CharlesChiuGit Using the `get` function  _might_ be a better option, suggestions for improvement are welcome 👍 

**THIS COMMIT MAY REQUIRE CHECKS & REVIEWS BEFORE MERGING. Though I've doubled-checked the code, there's still possibility that I missed some changes.**

### CODE REVIEW

**Comment by CharlesChiuGit** [comment]
Wow, this looks way nicer! Thanks for refactoring it!

**Comment by ayamir** [comment]
I think the `use_icon` flag is misunderstanding. Icons in `lualine` and `nvim-tree` and `telescope`'s candidate files are still here though I have set it to `false`. So maybe you need to config for every corner using icons to keep it intuitive and coherent.

**Comment by ayamir** [comment]
Additionally, there is a inconspicuous bug in `telescope`.
image
It can be reproduced by `<C-p>` and `<C-n>` repeatly in `telescope`.

**Comment by ayamir** [comment]
In fact, I think configurable `use_icons` is not a good idea because it hurts the complete user experience.

**Review by ayamir — APPROVED** [review]

**Comment by Jint-lzxy** [comment]
Indeed. I have also fixed relevant issues, now `Telescope` should work correctly:

https://user-images.githubusercontent.com/50296129/196016704-e33c3680-b5a6-4586-a12b-24d726b281d2.mp4

File: lua/modules/ui/icons.lua [hunk 1]
```diff
@@ -1,187 +1,206 @@
...
+		Bug = "",
+		Calendar = "",
+		Check = "",
+		ChevronRight = "",
+		Circle = "",
+		Close = "",
+		Close_alt = "",
+		CloudDownload = "",
+		Comment = "",
+		CodeAction = "",
```
[the remark this answers is not in the document]
Comment (Jint-lzxy) [reply lua/modules/ui/icons.lua:86]: Fixed at #289 👍

[the remark this answers is not in the document]
Comment (CharlesChiuGit) [reply lua/modules/ui/icons.lua:86]: Nice!

**Pull request merged**